lib/gtfs/timepoint.ex

```diff
@@ -0,0 +1,38 @@
+defmodule Gtfs.Timepoint do
+  alias Gtfs.Csv
+
+  @type id :: String.t()
+
+  @type t :: %__MODULE__{
+          id: id(),
+          name: String.t() | nil
+        }
+
+  @type timepoints_by_id :: %{id() => t()}
+
+  @enforce_keys [
+    :id
+  ]
+
+  @derive Jason.Encoder
+
+  defstruct [
+    :id,
+    name: nil
+  ]
+
+  @spec from_csv_row(Csv.row()) :: t()
+  def from_csv_row(row) do
+    %__MODULE__{
+      id: row["checkpoint_id"],
+      name: row["checkpoint_name"]
+    }
+  end
+
+  @spec timepoint_for_id(timepoints_by_id(), id()) :: t() | nil
+  def timepoint_for_id(_timepoints_by_id, nil), do: nil
+
+  def timepoint_for_id(timepoints_by_id, id) do
+    Map.get(timepoints_by_id, id, %__MODULE__{id: id})
```

Comment:
Instead of my other comments about the frontend handling a missing name, what do you think about creating a default `%__MODULE__{id: id, name: id}`, where the id is used as the name for the couple timepoints that we're missing, so we can depend on the `name` always being present?

Reply:
At least for now we've said that we don't want to display a tooltip if there is no name.